src/utils/generator.js

```diff
@@ -157,25 +158,7 @@ export class Generator {
           desc.get || object.$Realm.intrinsics.undefined,
           desc.set || object.$Realm.intrinsics.undefined,
         ],
-        buildNode: ([objectNode, valueNode, getNode, setNode]) => {
-          let descProps = [];
-          descProps.push(t.objectProperty(t.identifier("enumerable"), t.booleanLiteral(!!desc.enumerable)));
-          descProps.push(t.objectProperty(t.identifier("configurable"), t.booleanLiteral(!!desc.configurable)));
-          if (!desc.get && !desc.set) {
-            descProps.push(t.objectProperty(t.identifier("writable"), t.booleanLiteral(!!desc.writable)));
-            descProps.push(t.objectProperty(t.identifier("value"), valueNode));
-          } else {
-            descProps.push(t.objectProperty(t.identifier("get"), getNode));
-            descProps.push(t.objectProperty(t.identifier("set"), setNode));
-          }
-          return t.expressionStatement(
-            t.callExpression(this.preludeGenerator.memoizeReference("Object.defineProperty"), [
-              objectNode,
-              t.stringLiteral(key),
-              t.objectExpression(descProps),
-            ])
-          );
-        },
+        buildNode: (_, context: SerializationContext) => context.emitDefinePropertyBody(object, key, desc),
```

Comment:
So sad that Flow doesn't catch this: entries in this.body must have a buildNode field that are functions with a non-void return type, while context.emitDefinePropertyBody is specified to return void. @avikchaudhuri